agent/command/git_push.go

```diff
@@ -81,8 +82,12 @@ func (c *gitPush) Execute(ctx context.Context, comm client.Communicator, logger
 	}
 
 	// get commit information
+	cloneMethod := distro.CloneMethodOAuth
+	if conf.Distro != nil {
+		cloneMethod = conf.Distro.CloneMethod
+	}
```

Comment:
Can we put L85-88 in a method on `internal.TaskConfig` so that we always consistently return the proper OAuth default across git commands?

Reply:
Can we also add a test to verify that the git commands no longer depend on the distro clone method?

Reply:
Sure, moved to a method and updated existing unit tests.